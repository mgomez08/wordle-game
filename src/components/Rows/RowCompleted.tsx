import checkLetter from "../../helpers/checkLetter";
import { Box } from "../Box/Box";

import style from "./row.module.scss";
interface RowCompletedProps {
  word: string;
  solution: string;
}

export const RowCompleted = ({ word, solution }: RowCompletedProps) => {
  const ROWS: number = 5;
  const arrayRows = Array.from(Array(ROWS));
  return (
    <div className={style.row}>
      {arrayRows.map((_, index) => (
        <Box
          key={index}
          value={word[index]}
          status={checkLetter(word[index], index, solution)}
        />
      ))}
    </div>
  );
};
