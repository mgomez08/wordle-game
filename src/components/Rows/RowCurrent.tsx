import { Box } from "../Box/Box";
import styles from "./row.module.scss";

interface RowCurrentProps {
  word: string;
}

export const RowCurrent = ({ word }: RowCurrentProps) => {
  const wordArray = word.split("");
  return (
    <div className={styles.row}>
      {wordArray.map((letter, index) => (
        <Box key={index} value={letter} status="edit" />
      ))}
      {Array.from(Array(5 - wordArray.length)).map((letter, index) => (
        <Box key={index} value={""} status="edit" />
      ))}
    </div>
  );
};
