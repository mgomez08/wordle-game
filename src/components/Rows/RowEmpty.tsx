import { Box } from "../Box/Box";
import styles from "./row.module.scss";

export const RowEmpty = () => {
  const ROWS: number = 5;
  const arrayRows = Array.from(Array(ROWS));
  return (
    <div className={styles.row}>
      {arrayRows.map((_, index) => (
        <Box key={index} value="" status="empty" />
      ))}
    </div>
  );
};
