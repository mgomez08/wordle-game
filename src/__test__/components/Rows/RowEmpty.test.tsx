import { mount } from "enzyme";
import { RowEmpty } from "../../../components/Rows/RowEmpty";

describe("Tests for <RowEmpty", () => {
  const rowempty = mount(<RowEmpty />);

  test("Render RowEmpty component", () => {
    expect(rowempty.length).toEqual(1);
  });

  test("RowEmpty component should have correct number of children (boxes)", () => {
    expect(rowempty.find("Box").length).toEqual(5);
  });

  test("RowEmpty component should have class (.row)", () => {
    expect(rowempty.find(".row").length).toEqual(1);
  });

  test("First Box children should have correct class (.empty)", () => {
    expect(rowempty.find("Box").first().find(".empty").length).toEqual(1);
  });

  test("First Box children should haven't content", () => {
    expect(rowempty.find("Box").first().text()).toEqual("");
  });

  test("RowEmpty component with Snapshot", () => {
    expect(rowempty).toMatchSnapshot();
  });
});
