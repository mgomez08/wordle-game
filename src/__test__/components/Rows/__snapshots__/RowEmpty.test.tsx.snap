// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RowEmpty RowEmpty component with Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="empty"
  />
  <div
    className="empty"
  />
  <div
    className="empty"
  />
  <div
    className="empty"
  />
  <div
    className="empty"
  />
</div>
`;
